Automattic/wp-calypso
# Frictionless Email Subscriptions: Add new user creation to email subscription flow

Related to https://github.com/Automattic/martech/issues/3098

## Proposed Changes

* Create a new user upon initial rendering of the email subscription flow
* Use `redirect_to` query param and redirect to said url if new user creation is successful
* Add `subscription=true` query param to redirect url
* Show passwordless signup form with email query param autofilled if new user creation is unsuccessful ( Note: passwordless signup form isn't completely functional yet. Will be handled in a follow-up PR )

![2024-06-25 21 21 23](https://github.com/Automattic/wp-calypso/assets/5414230/12d1b1e0-5bfc-4356-9811-dda31940797b)

## Why are these changes being made?


When someone subscribes to an email campaign on a landing page like wordpress.com/learn, we'd like to manage outbound communication PCYsg-6j2-p2. Guides, however, requires a user_id and a WordPress account.

In the past, if the user wanted to subscribe on a landing page and wasn't already logged in, they'd be redirected to the /log-in page. If they didn't have an account, they'd be prompted to create one. Once completed, they'd be redirected to the original landing page.

To polish this experience, we're creating a custom email subscription "signup" flow. For the happy path, the user should be able to subscribe with one click, and the technical implementation should look something like p5uIfZ-f11-p2#comment-22793.

## Testing Instructions



* Check out this branch
* Set the `signup/email-subscription-flow` flag to true in `config/development.json`
* Start local dev
* Navigate to `/start/email-subscription/subscribing-email/subscribe?user_email={YOUR_NEW_EMAIL}&redirect_to=wordpress.com%2Flearn&mailing_list=test` Note that the query params are required.
* Verify that a loading screen is initially rendered
* Replace the email query param with the email of a brand new user
* Verify that a new user is created when the page is redirected
* Revisit `/start/email-subscription/subscribing-email/subscribe?user_email={YOUR_NEW_EMAIL}&redirect_to=wordpress.com%2Flearn&mailing_list=test` to emulate an email subscription for an existing user. We will subscribe them to guides automatically behind the scenes in a future PR. They should also be redirected to the `redirect_to` query param.

## Pre-merge Checklist
